odoo/tools/pdf.py

```diff
@@ -19,3 +49,302 @@ def merge_pdf(pdf_data):
     merged_pdf = _buffer.getvalue()
     _buffer.close()
     return merged_pdf
+
+
+class OdooPdfFileWriter(PdfFileWriter):
+
+    def __init__(self, pdf_content=None):
+        """
+        Init the reader, with an optional pdf_content
+        :param pdf_content: if given, will initialise the reader with the pdf content.
+        """
+        super().__init__()
+        self.pdf_content = b""
+        self.stack = []
+        if pdf_content is not None:
+            self.from_binary(pdf_content)
+
+    def from_binary(self, pdf_content):
+        """
+        Initialise the writer with the content from the provided bytes
+        :param pdf_content: the pdf content in bytes
+        """
+        self.pdf_content = pdf_content
+        reader_buffer = io.BytesIO(pdf_content)
+        reader = PdfFileReader(reader_buffer)
+        self.cloneDocumentFromReader(reader)
```

Comment:
IMO, you should follow the same behavior as the original class and make this part in `_add_facturx_attachments` and add the pdf-a parameter:
```
buffer = io.BytesIO(pdf_content)
reader = PdfFileReader(buffer)
writer = OdooPdfFileWriter()
writer.cloneDocumentFromReader(reader)
writer.enablePDFA(title="...", subject="...")
writer.write(...) # PDF-A should be handled when writing the final content
```